types/node/zlib.d.ts

```diff
@@ -2,16 +2,46 @@ declare module "zlib" {
     import * as stream from "stream";
 
     interface ZlibOptions {
-        flush?: number; // default: zlib.constants.Z_NO_FLUSH
-        finishFlush?: number; // default: zlib.constants.Z_FINISH
-        chunkSize?: number; // default: 16*1024
+        /**
+         * @default constants.Z_NO_FLUSH
+         */
+        flush?: number;
+        /**
+         * @default constants.Z_FINISH
+         */
+        finishFlush?: number;
+        /**
+         * @default 16*1024
+         */
+        chunkSize?: number;
         windowBits?: number;
         level?: number; // compression only
         memLevel?: number; // compression only
         strategy?: number; // compression only
         dictionary?: Buffer | NodeJS.TypedArray | DataView | ArrayBuffer; // deflate/inflate only, empty dictionary by default
     }
 
+    interface BrotliOptions {
+        /**
+         * @default constants.BROTLI_OPERATION_PROCESS
+         */
+        flush?: number;
+        /**
+         * @default constants.BROTLI_OPERATION_FINISH
+         */
+        finishFlush?: number;
+        /**
+         * @default 16*1024
+         */
+        chunkSize?: number;
+        params?: {
+            /**
+             * Each key is a `constants.BROTLI_*` constant.
+             */
+            [key: number]: boolean | number;
```

Comment:
Maybe we should name the properties here and use the correct type per option? Not sure if this will work if users use the constants provided.

Reply:
Tried that, TS 2.1 says no, especially since I can't refer to the `constants` this would be a giant mess, if you can figure out how to make it work in a maintainable manner I'll gladly add it.